### PR TITLE
Adds rule to check if each panel has valid units assigned

### PR DESCRIPTION
**Pull request by mshahzeb**
Reference issue:
https://github.com/grafana/dashboard-linter/issues/51

This rule checks that each panel must have a valid unit assigned.
The unit is extracted from dashboard json in the following manner:
- Parse overrides to check if any override unit is present
- If no overrride unit - then extract standard option unit
- Check if unit is defined and is valid

Valid units are extracted from here:
https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/valueFormats/categories.ts

Considerations:
- For a panel with multiple fields and panels - the last override unit will be picked

Sample output:
image

### CODE REVIEW

**Comment by mshahzeb** [comment]
Since the override value is of dynamic type - It can be an int or a string.

I am getting an error in case when a dashboard has mixed int and string values in overrides.
I am working on a fix for that too.

File: lint/rule_panel_units.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package lint
+
+import "fmt"
+
+func NewPanelUnitsRule() *PanelRuleFunc {
+	validUnits := []string{
+		// Enumerated from: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/valueFormats/categories.ts
+
+		// Misc
+		"none", "string",
```
Comment (mshahzeb) [inline lint/rule_panel_units.go:10]: Should we leave none as an allowed unit to be set if no other unit makes sense?

Or remove this and let the user add a lint exclusion in that case?

Comment (rgeyer) [reply lint/rule_panel_units.go:10]: I wasn't sure the behavior for a default dashboard/panel, so I just tested it.

When a new panel is created, with no config, no unit is set. Literally the `unit` property is absent in `fieldConfig`.

In the dropdown/picker for units, `none` does not appear to be an option. If you type it in manually, you can set it as a "custom" unit.

For this reason, I would say this shouldn't be an allowed valid unit, and if users choose to explicitly specify "none" they should also create a lint exclusion rule.

**Review by rgeyer — APPROVED** [review]
LGTM, with the one bit of feedback on the "none" unit type.

Great work. Thank you!

**Pull request merged**